src/lib.rs

```diff
@@ -828,159 +834,74 @@ impl<R: SeedableRng> FromEntropy for R {
     }
 }
 
-/// The standard RNG. The PRNG algorithm in `StdRng` is chosen to be efficient
-/// on the current platform, to be statistically strong and unpredictable
-/// (meaning a cryptographically secure PRNG).
+
+/// DEPRECATED: use [`SmallRng`] instead.
 ///
-/// The current algorithm used on all platforms is [HC-128].
+/// Create a weak random number generator with a default algorithm and seed.
 ///
-/// Reproducibility of output from this generator is however not required, thus
-/// future library versions may use a different internal generator with
-/// different output. Further, this generator may not be portable and can
-/// produce different output depending on the architecture. If you require
-/// reproducible output, use a named RNG, for example [`ChaChaRng`].
+/// It returns the fastest `Rng` algorithm currently available in Rust without
+/// consideration for cryptography or security. If you require a specifically
+/// seeded `Rng` for consistency over time you should pick one algorithm and
+/// create the `Rng` yourself.
 ///
-/// [HC-128]: prng/hc128/struct.Hc128Rng.html
-/// [`ChaChaRng`]: prng/chacha/struct.ChaChaRng.html
-#[derive(Clone, Debug)]
-pub struct StdRng(Hc128Rng);
-
-impl RngCore for StdRng {
-    #[inline(always)]
-    fn next_u32(&mut self) -> u32 {
-        self.0.next_u32()
-    }
-
-    #[inline(always)]
-    fn next_u64(&mut self) -> u64 {
-        self.0.next_u64()
-    }
-
-    fn fill_bytes(&mut self, dest: &mut [u8]) {
-        self.0.fill_bytes(dest);
-    }
-
-    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-        self.0.try_fill_bytes(dest)
-    }
-}
-
-impl SeedableRng for StdRng {
-    type Seed = <Hc128Rng as SeedableRng>::Seed;
-
-    fn from_seed(seed: Self::Seed) -> Self {
-        StdRng(Hc128Rng::from_seed(seed))
-    }
-
-    fn from_rng<R: RngCore>(rng: R) -> Result<Self, Error> {
-        Hc128Rng::from_rng(rng).map(StdRng)
-    }
+/// This will seed the generator with randomness from `thread_rng`.
+///
+/// [`SmallRng`]: rngs/struct.SmallRng.html
+#[deprecated(since="0.5.0", note="removed in favor of SmallRng")]
+#[cfg(feature="std")]
+pub fn weak_rng() -> XorShiftRng {
```

Comment:
Diff is messed up; this didn't change (only `random` pasted below and 2 Rngs removed)